extensions/pkg/controller/operatingsystemconfig/reconciler.go

```diff
@@ -195,19 +210,87 @@ func (r *reconciler) delete(ctx context.Context, osc *extensionsv1alpha1.Operati
 	return reconcile.Result{}, nil
 }
 
+func (r *reconciler) migrate(ctx context.Context, osc *extensionsv1alpha1.OperatingSystemConfig) (reconcile.Result, error) {
+	hasFinalizer, err := extensionscontroller.HasFinalizer(osc, FinalizerName)
+	if err != nil {
+		r.logger.Error(err, "Could not instantiate finalizer deletion")
+		return reconcile.Result{}, err
+	}
+	if !hasFinalizer {
+		r.logger.Info("Deleting operating system config causes a no-op as there is no finalizer.", "osc", osc.Name)
+		return reconcile.Result{}, nil
+	}
+
+	if err := r.updateStatusProcessing(ctx, osc, gardencorev1beta1.LastOperationTypeDelete, "Deleting the operating system config"); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if err := r.actuator.Delete(ctx, osc); err != nil {
```

Comment:
Shouldn't this be `r.actuator.Migrate()`

Reply:
Sure it must be, thanks!